.github/workflows/pipeline.yml

```
@@ -0,0 +1,195 @@
+    needs: [unittest_old, unittest_matrix, pip_check, coverage]
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    if: (github.event.pull_request.user.login == 'dependabot[bot]' || github.event.pull_request.user.login == 'pre-commit-ci[bot]')
+    steps:
+      - name: Enable auto-merge for bot PRs
+        run: gh pr merge --auto --squash "$PR_URL"
+        env:
+          PR_URL: ${{github.event.pull_request.html_url}}
+          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)